### PR TITLE
Loop dashboard new template

### PR DESCRIPTION
**Pull request by Abhishek-lodha**
Sustainability bug on recent graph page fixed.

### CODE REVIEW

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 1]
```diff
@@ -145,66 +187,53 @@ function total_static_data() {
             total_farmer_share += json_data['total_transportation_cost'][i]['farmer_share__sum'];
         }
 
+        var total_gaddidar_contribution = json_data['total_gaddidar_contribution'];
+        // var total_gaddidar_contribution=0;
+        // for (var i = 0; i < json_data['total_gaddidar_contribution'].length; i++) {
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: remove this extra commented code if not required

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 2]
```diff
@@ -145,66 +187,53 @@ function total_static_data() {
...
 }
 
 
 //To plot and show data for recents graphs on home page
 function plot_cards_data() {
-    var avg = get_average(); // Retunts [avg_volume, active_farmers, avg_amount]
+    var avg = get_average(); // Retunts [avg_vol, active_farmers, avg_amt, active_clusters,avg_gaddidar_share]
     var avg_vol = avg[0];
+    var avg_gaddidar_contribution = avg[4];
     var kg = "Kg";
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: Can this initialisation be done once only ? Already done at line 204

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 3]
```diff
@@ -332,16 +400,20 @@ function get_cpk(avg_vol) {
         k++;
     }
     while (j < transportation.length && today < new Date(transportation[j]['date'])) {
-        temp += transportation[j]['transportation_cost__sum']; // - transportation[j]['farmer_share__sum'];
-        f_share += transportation[j]['farmer_share__sum'];
+        temp += parseFloat(transportation[j]['transportation_cost__sum']); // - transportation[j]['farmer_share__sum'];
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: why temp ? what does it refer to ?

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 4]
```diff
@@ -456,21 +534,22 @@ function change_graph(parameter) {
...
-            update_graphs_gaddidar_wise(parameter);
-            gaddidar = false;
-        }
-
-    }
-    if (selected_tab == "crop") {
+        // if (gaddidar) {
+        update_graphs_gaddidar_wise(selected_parameter);
+        // gaddidar = false;
+        // }
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: remove commented lines

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 5]
```diff
@@ -597,7 +664,12 @@ function set_filterlistener() {
             $('#aggregator_payment_details').hide();
             $('#payments_to_date').prop('disabled', false);
             var from_date = new Date(new Date(start_date));
-            $('#payments_to_date').val(from_date.getFullYear() + "-" + (from_date.getMonth() + 1) + "-" + (from_date.getDate() + 15));
+            var daysInMonth = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+            if (from_date.getDate() >= 16 && daysInMonth[from_date.getMonth()] != (from_date.getDate() + 15)) {
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: isn't there any ready made function or library available that we can use instead of handling date thing at our end ?

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 6]
```diff
@@ -725,18 +829,16 @@ function get_data() {
     });
 
     if (Date.parse(start_date) > Date.parse(end_date)) {
-        //$('.modal-trigger').leanModal();
         $('#modal1').openModal();
     } else {
         $(".button-collapse1").sideNav('hide');
         get_data_for_bar_graphs(start_date, end_date, aggregator_ids, crop_ids, mandi_ids, gaddidar_ids);
         get_data_for_line_graphs(start_date, end_date, aggregator_ids, crop_ids, mandi_ids, gaddidar_ids);
-        // genterate_payment_sheet(start_date, end_date);
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: remove this line

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 7]
```diff
@@ -910,48 +1012,54 @@ function aggregator_graph(container, axis, axis_names, axis_parameter, values, v
...
-        var drilldown_index = values.indexOf(json_data[i][values_parameter].toString());
-        var index = axis.indexOf(json_data[i][axis_parameter].toString());
-
-        drilldown['series'][index]['data'][drilldown_index]['y'] += json_data[i][parameter]
-
-        series[0]['data'][index]['y'] += json_data[i][parameter];
-    }
+    // for (var i = 0; i < json_data.length; i++) {
+    //     var drilldown_index = values.indexOf(json_data[i][values_parameter].toString());
+    //     var index = axis.indexOf(json_data[i][axis_parameter].toString());
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: Please clean this up if not required

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 8]
```diff
@@ -910,48 +1012,54 @@ function aggregator_graph(container, axis, axis_names, axis_parameter, values, v
...
-                break;
-            }
-        }
     }
 
-
+    // for (var i = 0; i < axis.length; i++) {
+    //     drilldown['series'][i]['data'].sort(function(a, b) {
+    //         return b['y'] - a['y'];
+    //     });
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: clean up this code if not required for future reference

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 9]
```diff
@@ -975,38 +1083,74 @@ function transport_cost_graph(container, axis, axis_names, axis_parameter, value
...
-        drilldown['series'][index * 2 + 1]['data'][drilldown_index][1] += json_data[i]['farmer_share__sum']
-        data_for_sorting[index]['cost'] += json_data[i]['transportation_cost__sum']
-        data_for_sorting[index]['cost_recovered'] += json_data[i]['farmer_share__sum'];
-    }
+    // var json_data_length = json_data.length;
+    // for (var i = 0; i < json_data_length; i++) {
+    //     var index = axis.indexOf(json_data[i][axis_parameter].toString());
+    //     var drilldown_index = values.indexOf(json_data[i][values_parameter].toString());
+    //     drilldown['series'][index * 2]['data'][drilldown_index][1] += json_data[i]['transportation_cost__sum'];
+    //     drilldown['series'][index * 2 + 1]['data'][drilldown_index][1] += json_data[i]['farmer_share__sum'];
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: Clean up ?

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 10]
```diff
@@ -1016,29 +1160,31 @@ function transport_cost_graph(container, axis, axis_names, axis_parameter, value
...
-
     plot_drilldown(container, series, drilldown, false);
 }
 
 //Cpk and Spk on analytics page is being plotted from this
 function cpk_spk_graph(container, axis, axis_names, axis_parameter, values, values_names, values_parameter, json_data) {
-    var vol_stats = json_data.aggregator_mandi;
+    //Not considering crops filter while calculating cpk and spk
+    // var vol_stats = json_data.transactions_details_without_crops;
+    //TODO: can merge vol_stats and gaddidar_shre_stats as the data is without applying crop filters
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: Remove extra comments and TODO statement if already done

Comment (Abhishek-lodha) [inline dg/media/app_dashboards/js/loop_dashboard.js]: Need those.

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 11]
```diff
@@ -867,38 +950,57 @@ function totals() {
...
+
+    total_recovered += gaddidar_share;
+    total_cost += volume_without_crop_gaddidar_filter * AGGREGATOR_INCENTIVE_PERCENTAGE;
 
     $("#aggregator_volume").text("Volume: " + parseFloat(total_volume).toFixed(0) + " Kg");
     $("#aggregator_amount").text("amount: " + "₹ " + parseFloat(total_amount).toFixed(0));
     $("#aggregator_visits").text("visits: " + total_visits);
     $("#aggregator_cpk").text("SPK/CPK : " + spk + "/" + cpk);
-    $("#aggregator_cost").text("Recovered/Total : " + total_recovered + "/" + total_cost);
+    $("#aggregator_cost").text("Recovered/Total : " + "₹ " + total_recovered.toFixed(2) + "/" + " ₹ " + total_cost.toFixed(2));
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: use initialised variable instead of Rs symbol

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 12]
```diff
@@ -867,38 +950,57 @@ function totals() {
...
+        volume_without_crop_gaddidar_filter += gaddidar_contribution[i]['quantity__sum'];
+    }
+
+    var cpk = ((total_cost + volume_without_crop_gaddidar_filter * AGGREGATOR_INCENTIVE_PERCENTAGE) / volume_without_crop_gaddidar_filter).toFixed(2);
+    var spk = ((total_recovered + gaddidar_share) / volume_without_crop_gaddidar_filter).toFixed(2);
+
+    total_recovered += gaddidar_share;
+    total_cost += volume_without_crop_gaddidar_filter * AGGREGATOR_INCENTIVE_PERCENTAGE;
 
     $("#aggregator_volume").text("Volume: " + parseFloat(total_volume).toFixed(0) + " Kg");
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: Why hard coding "Kg" when you have already initialised somewhere

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 13]
```diff
@@ -2809,7 +2818,8 @@ function aggregator_payment_sheet(data_json, aggregator) {
 
     var aggregator_payment = payments_data.aggregator_data;
     var transport_payment = payments_data.transportation_data;
-    var gaddidar_payment = payments_data.gaddidar_data;
+    var gaddidar_contribution_data = payments_data.gaddidar_data;
+    // var gaddidar_payment = payments_data.gaddidar_data;
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: remove this line

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 14]
```diff
@@ -2809,7 +2818,8 @@ function aggregator_payment_sheet(data_json, aggregator) {
...
 
     var sno = 1;
     var str1 = "Rs. ";
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: Why Rs. instead of symbol when symbol is everywhere else ? Also, Can global variable be used ?

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 15]
```diff
@@ -2851,31 +2861,55 @@ function aggregator_payment_sheet(data_json, aggregator) {
...
-            var net_payment = quantites[i][j] * 0.25 - gaddidar_amount[i][j] + transport_cost[i][j] - farmer_share[i][j];
+            var net_payment = quantites[i][j] * AGGREGATOR_INCENTIVE_PERCENTAGE + transport_cost[i][j] - farmer_share[i][j];
+            //  - gaddidar_amount[i][j];
 
-            data_set.push([sno, dates[i], mandis[i][j], (quantites[i][j]).toString().concat(" Kg"), farmers[i][j], (quantites[i][j] * 0.25).toFixed(2), transport_cost[i][j], farmer_share[i][j], (gaddidar_amount[i][j]).toFixed(0), net_payment]);
+            data_set.push([sno, dates[i], mandis[i][j], (quantites[i][j]).toString().concat(" Kg"), farmers[i][j], (quantites[i][j] * AGGREGATOR_INCENTIVE_PERCENTAGE).toFixed(2), transport_cost[i][j], farmer_share[i][j], 0, net_payment]);
             sno += 1;
-            total_volume += quantites[i][j];
-            total_payment += net_payment;
+            // total_volume += quantites[i][j];
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: remove commented lines

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 16]
```diff
@@ -2851,31 +2861,55 @@ function aggregator_payment_sheet(data_json, aggregator) {
...
         }
     }
 
     var total_volume = 0;
     var total_payment = 0;
     for (var i = 0; i < dates.length; i++) {
         for (var j = 0; j < mandis[i].length; j++) {
-            var net_payment = quantites[i][j] * 0.25 - gaddidar_amount[i][j] + transport_cost[i][j] - farmer_share[i][j];
+            var net_payment = quantites[i][j] * AGGREGATOR_INCENTIVE_PERCENTAGE + transport_cost[i][j] - farmer_share[i][j];
+            //  - gaddidar_amount[i][j];
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: remove commented line

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 17]
```diff
@@ -2968,28 +3002,36 @@ function aggregator_payment_sheet(data_json, aggregator) {
...
         }
 
     });
-    // $("#table4").hide();
-    // $("#table4").tableExport([],"transport");
-
+}
 
+function getFormattedDate(aggregator_id) {
+    var monthNames = ["Jan", "Feb", "Mar", "Apr", "May", "June",
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: Again, do we really need to use monthNames variable ?

File: dg/media/app_dashboards/js/loop_dashboard.js [hunk 18]
```diff
@@ -3280,92 +3331,18 @@ function plot_solid_guage(container, minimum, present, target) {
...
-//     }
-
-//     plot_drilldown(container, series, drilldown);
-// }
-
-
-/* Formatting function for row details - modify as you need */
+function change_language(lang) {
+    language = lang;
+    // get_filter_data(language);
```
Comment (lokeshgarg92) [inline dg/media/app_dashboards/js/loop_dashboard.js]: remove line

File: dg/templates/app_dashboards/loop_base.html [hunk 19]
```diff
@@ -31,13 +31,13 @@
...
-<script src="{% static 'app_dashboards/js/materialize.js' %}"></script>
+<!-- <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script> -->
+<script src="{% static 'social_website/scripts/libs/external/jquery-1.11.1.min.js' %}"></script>
+<script src="{% static 'app_dashboards/js/materialize.min.js' %}"></script>
 <script src="{% static 'app_dashboards/js/jquery.sparkline.min.js' %}"></script>
 <script src="{% static 'app_dashboards/datatablejs_media/js/jquery.dataTables.min.js' %}"></script>
-<script src="{% static 'app_dashboards/js/dataTables.tableTools.js' %}"></script>
-<script src="{% static 'app_dashboards/js/tableExporter.min.js' %}"></script>
+<script src="{% static 'social_website/scripts/libs/external/dataTables.tableTools.js' %}"></script>
+<!-- <script src="{% static 'app_dashboards/js/tableExporter.min.js' %}"></script> -->
```
Comment (lokeshgarg92) [inline dg/templates/app_dashboards/loop_base.html]: remove line

File: dg/templates/app_dashboards/loop_base.html [hunk 20]
```diff
@@ -31,13 +31,13 @@
 
 <!-- Scripts here -->
 <!-- Placed at the end of file for loading faster -->
-<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
-<!-- <script src="{% static 'app_dashboards/js/jquery.min.js' %}"></script> -->
-<script src="{% static 'app_dashboards/js/materialize.js' %}"></script>
+<!-- <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script> -->
```
Comment (lokeshgarg92) [inline dg/templates/app_dashboards/loop_base.html]: since there is 1.12.4 already being used on website, can we use same library instead of 1.11 jquery ?

File: dg/templates/app_dashboards/loop_dashboard.html [hunk 21]
```diff
@@ -319,19 +365,19 @@ <h5 id="recent_sustainability_card"></h5>
...
-	<div class="center">
-
-		<div id="modal1" class="modal">
-			<div class="modal-content">
-				<p>Invalid Date Range</p>
-			</div>
-			<div class="modal-footer">
-				<a href="#!" class=" modal-action modal-close waves-effect waves-black btn-flat">OK</a>
-			</div>
+<!-- <div class="center"> -->
```
Comment (lokeshgarg92) [inline dg/templates/app_dashboards/loop_dashboard.html]: remove

File: dg/templates/app_dashboards/loop_dashboard.html [hunk 22]
```diff
@@ -319,19 +365,19 @@ <h5 id="recent_sustainability_card"></h5>
...
+	<div id="modal1" class="modal">
+		<div class="modal-content">
+			<h4>Invalid Date Range</h4>
+			<p>Please select a valid date range. <br> &lt;From&gt; cannot be greater than &lt;To&gt; date.</p>
+		</div>
+		<div class="modal-footer">
+			<a href="#!" class="modal-action modal-close waves-effect waves-black btn-flat">OK</a>
 		</div>
 	</div>
+<!-- </div> -->
```
Comment (lokeshgarg92) [inline dg/templates/app_dashboards/loop_dashboard.html]: remove line

File: dg/templates/app_dashboards/loop_dashboard.html [hunk 23]
```diff
@@ -469,66 +505,44 @@ <h5 class="center-align">SPK</h5>
...
-			<br>
+			<br><br><br>
 			<hr/>
 
 			<div class="row center">
 				<h3>Payment Sheets</h3>
 			</div>
 			<div class="row center">
-				<div class="col s4" style="overflow-y:auto;">
+				<!-- <div class="col s4" style="overflow-y:auto;"> -->
```
Comment (lokeshgarg92) [inline dg/templates/app_dashboards/loop_dashboard.html]: remove line

File: dg/templates/app_dashboards/loop_dashboard.html [hunk 24]
```diff
@@ -469,66 +505,44 @@ <h5 class="center-align">SPK</h5>
...
 					<table id="table2" class="stripe">
 					</table>
-				</div>
-				<p class="col s1">
-
-					<div class="col s3" style="overflow-y:auto;">
+					<br>
+					<hr/>
+				<!-- </div> -->
+				<!-- <p class="col s1"> -->
```
Comment (lokeshgarg92) [inline dg/templates/app_dashboards/loop_dashboard.html]: remove commented lines

File: dg/templates/app_dashboards/loop_dashboard.html [hunk 25]
```diff
@@ -469,66 +505,44 @@ <h5 class="center-align">SPK</h5>
...
+					<!-- <div class="col s3" style="overflow-y:auto;"> -->
 						<table id="table3" class="stripe"></table>
-					</div>
-					<p class="col s1">
-						<div class="col s3" style="overflow-y:auto;">
+						<br>
+						<hr/>
+					<!-- </div> -->
+					<!-- <p class="col s1"> -->
```
Comment (lokeshgarg92) [inline dg/templates/app_dashboards/loop_dashboard.html]: remove commented line

File: dg/templates/app_dashboards/loop_dashboard.html [hunk 26]
```diff
@@ -469,66 +505,44 @@ <h5 class="center-align">SPK</h5>
...
 							<table id="table4" class="stripe"></table>
-						</div>
+							<br>
+							<hr/>
+						<!-- </div> -->
 			</div>
 		</div>
 	</div>
 	<!-- <div class="row center">
 		<a id="download_payment_sheets" class="waves-effect waves-light grey darken-3 white-text btn modal-trigger" onclick="download_payments_data()">Download</a>
```
Comment (lokeshgarg92) [inline dg/templates/app_dashboards/loop_dashboard.html]: please cleanup this commented code

File: loop/models.py [hunk 27]
```diff
@@ -54,7 +61,9 @@ class District(LoopModel):
     district_name = models.CharField(max_length=50)
     state = models.ForeignKey(State)
     is_visible = models.BooleanField(default=True)
+
```
Comment (lokeshgarg92) [inline loop/models.py]: remove this extra line, district_name_en is part of models only

File: loop/urls.py [hunk 28]
```diff
@@ -31,17 +31,16 @@
...
-    url(r'^second_loop_page/', second_loop_page),
     url(r'^village_wise_data/', village_wise_data),
     url(r'^aggregator_wise_data/', aggregator_wise_data),
     url(r'^crop_wise_data/', crop_wise_data),
     url(r'^filter_data/', filter_data),
     url(r'^total_static_data/',total_static_data),
     url(r'^recent_graphs_data/',recent_graphs_data),
-    url(r'^farmer_count_aggregator_wise/',farmer_count_aggregator_wise),
-    url(r'^new_aggregator_wise_data/',new_aggregator_wise_data),
+    # url(r'^farmer_count_aggregator_wise/',farmer_count_aggregator_wise),
```
Comment (lokeshgarg92) [inline loop/urls.py]: remove this line

File: loop/urls.py [hunk 29]
```diff
@@ -31,17 +31,16 @@
...
+    url(r'^data_for_drilldown_graphs/',data_for_drilldown_graphs),
     url(r'^data_for_line_graph/',data_for_line_graph),
-    url(r'^data_for_time_chart/',data_for_time_chart),
+    # url(r'^data_for_time_chart/',data_for_time_chart),
```
Comment (lokeshgarg92) [inline loop/urls.py]: remove this line

File: loop/views.py [hunk 30]
```diff
@@ -51,19 +51,20 @@ def dashboard(request):
...
-    villages = Village.objects.extra(select={'village_name':'village_name_en'}).values('id', 'village_name')
-    crops = Crop.objects.extra(select={'crop_name':'crop_name_en'}).values('id', 'crop_name')
-    mandis = Mandi.objects.extra(select={'mandi_name':'mandi_name_en'}).values('id', 'mandi_name')
-    gaddidars = Gaddidar.objects.extra(select={'gaddidar_name':'gaddidar_name_en'}).values('id', 'gaddidar_name')
+    language = request.GET.get('language')
+    aggregators = LoopUser.objects.all().values('user__id', 'name', 'name_en')
+    villages = Village.objects.all().values('id', 'village_name', 'village_name_en')
+    crops = Crop.objects.all().values('id', 'crop_name')
+    cropsLang = Croplanguage.objects.values('crop__id', 'crop_name')
+    cropslanguage = [{'id': x['crop__id'],
```
Comment (lokeshgarg92) [inline loop/views.py]: user better variable instead of x

Comment (lokeshgarg92) [inline loop/views.py]: follow one convention either camelcase or underscore but make it crops_language at this place and wherever referred.

File: loop/views.py [hunk 31]
```diff
@@ -169,61 +170,121 @@ def total_static_data(request):
...
+    # total_volume_for_transport = CombinedTransaction.objects.aggregate(Sum('quantity'))
     total_repeat_farmers = len(CombinedTransaction.objects.values(
         'farmer').annotate(farmer_count=Count('farmer')).exclude(farmer_count=1))
     total_farmers_reached = len(
         CombinedTransaction.objects.values('farmer').distinct())
     total_cluster_reached = len(LoopUser.objects.all())
 
     # remove date from filter
-    total_transportation_cost = DayTransportation.objects.filter(date__gte="2016-06-01").values('date', 'user_created__id', 'mandi__id').annotate(
+    # filter(date__gte="2016-06-01")
```
Comment (lokeshgarg92) [inline loop/views.py]: remove extra line

File: loop/views.py [hunk 32]
```diff
@@ -169,61 +170,121 @@ def total_static_data(request):
     total_volume = CombinedTransaction.objects.all(
     ).aggregate(Sum('quantity'), Sum('amount'))
     # remove total_volume_for_transport after entering past data and make
-    # changes in js accordingly
-    total_volume_for_transport = CombinedTransaction.objects.filter(
-        date__gte="2016-06-01").aggregate(Sum('quantity'))
+    # changes in js accordingly (filter for june dates)
+    # total_volume_for_transport = CombinedTransaction.objects.aggregate(Sum('quantity'))
```
Comment (lokeshgarg92) [inline loop/views.py]: remove commented lines

File: loop/views.py [hunk 33]
```diff
@@ -169,61 +170,121 @@ def total_static_data(request):
...
     total_cluster_reached = len(LoopUser.objects.all())
 
     # remove date from filter
-    total_transportation_cost = DayTransportation.objects.filter(date__gte="2016-06-01").values('date', 'user_created__id', 'mandi__id').annotate(
+    # filter(date__gte="2016-06-01")
+    total_transportation_cost = DayTransportation.objects.values('date', 'user_created__id', 'mandi__id').annotate(
         Sum('transportation_cost'), farmer_share__sum=Avg('farmer_share'))
 
+    gaddidar_share = gaddidar_contribution_for_totat_static_data()
+    # gaddidar_share=calculate_gaddidar_share(None, None, None, None)
```
Comment (lokeshgarg92) [inline loop/views.py]: remove commented line

File: loop/models.py [hunk 34]
```diff
@@ -206,16 +222,26 @@ class Crop(LoopModel):
...
     class Meta:
         unique_together = ("crop_name",)
 
-
 post_save.connect(save_log, sender=Crop)
 pre_delete.connect(delete_log, sender=Crop)
 
+#############Crop name in multiple languages###############
+
+class Croplanguage(models.Model):
```
Comment (lokeshgarg92) [inline loop/models.py]: please change it to CropLanguage and regenerate the migrations and make the required changes wherever table is used. Convention must be followed in models

File: dg/media/app_dashboards/js/loop_dashboard.js (hunk 17 above)
Comment (Abhishek-lodha) [inline dg/media/app_dashboards/js/loop_dashboard.js]: yes. Date only returns month number

File: loop/views.py [hunk 35]
```diff
@@ -169,61 +170,121 @@ def total_static_data(request):
...
+                    sum += gso_gaddidar_date_aggregator[0]
+            except GaddidarShareOutliers.DoesNotExist:
+                pass
+        result.append({'date': CT['date'], 'user_created__id': CT['user_created_id'], 'gaddidar__id': CT[
+                      'gaddidar'], 'mandi__id': CT['mandi'], 'amount': sum, 'quantity__sum': CT['quantity__sum']})
+    return result
+
+
+def crop_language_data(request):
+    crops = Croplanguage.objects.filter(language=request.GET.get('language'))
```
Comment (lokeshgarg92) [inline loop/views.py]: Change the Croplanguage to CropLanguage after making required changes in models.py

File: loop/views.py [hunk 36]
```diff
@@ -169,61 +170,121 @@ def total_static_data(request):
...
-    filter_args = {}
-    if (start_date != ""):
-        filter_args["date__gte"] = start_date
-    if (end_date != ""):
-        filter_args["date__lte"] = end_date
-    farmers_count = CombinedTransaction.objects.filter(**filter_args).values(
-        'user_created__id').annotate(Count('farmer', distinct=True))
+    gaddidar_contribution = calculate_gaddidar_share(None, None, None, None)
+    # crops = Crop.objects.all().extra(
+    # select={'crop_name': 'crop_name_en'}).values('id', 'crop_name')
```
Comment (lokeshgarg92) [inline loop/views.py]: remove commented lines

File: loop/views.py [hunk 37]
```diff
@@ -232,64 +293,62 @@ def new_aggregator_wise_data(request):
...
-
-    crop_prices = CombinedTransaction.objects.filter(
-        **filter_args).annotate(crop__crop_name=F('crop__crop_name_en')).values('crop__crop_name','crop__id').annotate(Min('price'), Max('price'), Count('farmer', distinct=True))
 
+    transportation_cost_mandi = DayTransportation.objects.filter(**filter_transportation).values('date',
+                                                                                                 'mandi__id', 'user_created__id').annotate(Sum('transportation_cost'), farmer_share__sum=Avg('farmer_share'))
+
+    crop_prices = list(CombinedTransaction.objects.filter(
+        **filter_args).values('crop__crop_name', 'crop__id').annotate(Min('price'), Max('price'), Count('farmer', distinct=True)))
+    for i in crop_prices:
```
Comment (lokeshgarg92) [inline loop/views.py]: use better variable instead of i

File: loop/views.py [hunk 38]
```diff
@@ -232,64 +293,62 @@ def new_aggregator_wise_data(request):
...
+        try:
+            crop = Croplanguage.objects.get(crop=i['crop__id'])
```
Comment (lokeshgarg92) [inline loop/views.py]: CropLanguage instead of Croplanguage

File: loop/views.py [hunk 39]
```diff
@@ -311,71 +370,80 @@ def data_for_line_graph(request):
         filter_args["date__lte"] = end_date
         filter_transportation["date__lte"] = end_date
     filter_args["user_created__id__in"] = aggregator_ids
-    filter_args["crop__id__in"] = crop_ids
+    # filter_args["crop__id__in"] = crop_ids
```
Comment (lokeshgarg92) [inline loop/views.py]: remove line

File: loop/models.py [hunk 40]
```diff
@@ -206,16 +222,26 @@ class Crop(LoopModel):
     measuring_unit = models.CharField(max_length=20, default="kg")
     is_visible = models.BooleanField(default=True)
     crop_name_en = models.CharField(max_length=30, null=True)
```
Comment (lokeshgarg92) [inline loop/models.py]: We can remove this crop_name_en since a new table has been created now

File: loop/views.py [hunk 41]
```diff
@@ -387,26 +455,25 @@ def payments(request):
...
         **filter_args).annotate(mandi__mandi_name=F('mandi__mandi_name_en')).values('date', 'user_created__id', 'mandi__mandi_name').annotate(Sum('quantity'), Count('farmer', distinct=True)).annotate(gaddidar__commission__sum=Sum(F('gaddidar__commission') * F("quantity")))
 
     outlier_transport_data = DayTransportation.objects.filter(**filter_args).values(
         'date', 'mandi__id', 'user_created__id').annotate(Sum('transportation_cost'), farmer_share__sum=Avg('farmer_share'))
 
-    outlier_daily_data = CombinedTransaction.objects.filter(**filter_args).annotate(mandi__mandi_name=F('mandi__mandi_name_en'),gaddidar__gaddidar_name=F('gaddidar__gaddidar_name_en'),crop__crop_name=F('crop__crop_name_en')).values('date', 'user_created__id', 'mandi__mandi_name',
-                                                                                  'farmer__name', 'crop__crop_name', 'gaddidar__commission', 'price', 'gaddidar__gaddidar_name').annotate(Sum('quantity'))
+    outlier_daily_data = CombinedTransaction.objects.filter(**filter_args).annotate(mandi__mandi_name=F('mandi__mandi_name_en'), gaddidar__gaddidar_name=F('gaddidar__gaddidar_name_en'), crop__crop_name=F(
```
Comment (lokeshgarg92) [inline loop/views.py]: This is still using crop_name_en field from Crop table

File: training/models.py [hunk 42]
```diff
@@ -8,7 +8,7 @@
 # Create your models here.
     
 class TrainingUser(models.Model):
-    id = models.AutoField(primary_key=True)
+    id = models.AutoField (primary_key=True)
```
Comment (lokeshgarg92) [inline training/models.py]: undo this change

File: loop/views.py (hunk 38 above)
Comment (Abhishek-lodha) [inline loop/views.py]: MODEL NAME IS Croplanguage

File: dg/templates/app_dashboards/loop_dashboard.html [hunk 43]
```diff
@@ -30,6 +30,13 @@
 					<li id="payments_tab"><a class="white-text" onclick="hide_nav('payments')">Payments</a></li>
 				</ul>
 			</div>
+			<div class="col s2" class="initialized" style="float:right;">
+				<select id='languageID' onchange='change_language(this.value)'>
+					<!-- <option value="-1" disabled selected> Choose Language</option>> -->
```
Comment (lokeshgarg92) [inline dg/templates/app_dashboards/loop_dashboard.html]: remove commented line
